sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj

```diff
@@ -18,7 +18,7 @@
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\src\Azure.Core.csproj" />
-    <ProjectReference Include="..\..\..\keyvault\Azure.Security.KeyVault.Secrets\src\Azure.Security.KeyVault.Secrets.csproj" />
+    <PackageReference Include="Azure.Security.KeyVault.Secrets" />
```

Comment:
> To fix this I had to change Azure.Core.Test project to take a
project reference to keyvault.

Hm?

Why does this help? It stil results in Azure.Core package reference of version 1.0.0.

Reply:
Good point I'll keep digging to see if I can figure that out. For reference here are the errors we get without this change:
```
                   "D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj" (Restore target) (1) ->
                   (Restore target) -> 
                     D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj : error NU1106: Unable to satisfy conflicting requests for 'System.Threading.Tasks.Extensions': System.Threading.Tasks.Extensions (>= 4.5.2) (via package/Azure.Identity 1.0.0), System.Threading.Tasks.Extensions (>= 4.5.2) (via package/Microsoft.Bcl.AsyncInterfaces 1.0.0), System.Threading.Tasks.Extensions (>= 4.5.1) (via package/Moq 4.10.1), System.Threading.Tasks.Extensions (>= 4.5.2) (via package/System.Text.Json 4.6.0), System.Threading.Tasks.Extensions (>= 4.5.2) (via project/Azure.Core 1.0.0), System.Threading.Tasks.Extensions (>= 4.5.2) (via project/Azure.Security.KeyVault.Secrets 4.0.0), System.Threading.Tasks.Extensions (>= 4.5.1) (via package/Microsoft.AspNetCore.Server.Kestrel.Core 2.2.0), System.Threading.Tasks.Extensions (>= 4.5.1) (via package/System.IO.Pipelines 4.5.2) Framework: (.NETFramework,Version=v4.6.1)
                     D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj : error NU1106: Unable to satisfy conflicting requests for 'System.Buffers': System.Buffers (>= 4.4.0) (via package/System.Memory 4.5.3), System.Buffers (>= 4.5.0) (via package/System.Text.Json 4.6.0), System.Buffers (>= 4.5.0) (via project/Azure.Core 1.0.0), System.Buffers (>= 4.5.0) (via package/Azure.Core 1.0.0), System.Buffers (>= 4.5.0) (via package/Microsoft.Net.Http.Headers 2.2.0), System.Buffers (>= 4.5.0) (via package/Microsoft.AspNetCore.Http.Extensions 2.2.0), System.Buffers (>= 4.5.0) (via package/Microsoft.Net.Http.Headers 2.2.0), System.Buffers (>= 4.4.0) (via package/System.IO.Pipelines 4.5.2), System.Buffers (>= 4.4.0) (via package/System.IO.Pipelines 4.5.2) Framework: (.NETFramework,Version=v4.6.1)
                     D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj : error NU1106: Unable to satisfy conflicting requests for 'System.Numerics.Vectors': System.Numerics.Vectors (>= 4.4.0) (via package/System.Memory 4.5.3), System.Numerics.Vectors (>= 4.5.0) (via package/System.Text.Json 4.6.0), System.Numerics.Vectors (>= 4.5.0) (via project/Azure.Core 1.0.0), System.Numerics.Vectors (>= 4.5.0) (via package/Microsoft.AspNetCore.Server.Kestrel.Core 2.2.0), System.Numerics.Vectors (>= 4.5.0) (via package/Azure.Core 1.0.0) Framework: (.NETFramework,Version=v4.6.1)
                     D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj : error NU1106: Unable to satisfy conflicting requests for 'System.Runtime.CompilerServices.Unsafe': System.Runtime.CompilerServices.Unsafe (>= 4.5.2) (via package/System.Memory 4.5.3), System.Runtime.CompilerServices.Unsafe (>= 4.6.0) (via package/System.Text.Json 4.6.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.2) (via package/System.Threading.Tasks.Extensions 4.5.2), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.AspNetCore.Server.Kestrel.Core 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.2) (via package/System.Threading.Tasks.Extensions 4.5.2), System.Runtime.CompilerServices.Unsafe (>= 4.5.0) (via package/System.Threading.Tasks.Extensions 4.5.1), System.Runtime.CompilerServices.Unsafe (>= 4.5.2) (via package/System.Threading.Tasks.Extensions 4.5.2), System.Runtime.CompilerServices.Unsafe (>= 4.5.2) (via package/System.Threading.Tasks.Extensions 4.5.2), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.0) (via package/System.Threading.Tasks.Extensions 4.5.1), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0), System.Runtime.CompilerServices.Unsafe (>= 4.5.1) (via package/Microsoft.Extensions.Primitives 2.2.0) Framework: (.NETFramework,Version=v4.6.1)
                     D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj : error NU1106: Unable to satisfy conflicting requests for 'System.Diagnostics.DiagnosticSource': System.Diagnostics.DiagnosticSource (>= 4.6.0) (via project/Azure.Core 1.0.0), System.Diagnostics.DiagnosticSource (>= 4.5.0) (via package/Microsoft.AspNetCore.Hosting 2.2.0), System.Diagnostics.DiagnosticSource (>= 4.6.0) (via package/Azure.Core 1.0.0) Framework: (.NETFramework,Version=v4.6.1)
                     D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj : error NU1106: Unable to satisfy conflicting requests for 'System.Threading.Tasks.Extensions': System.Threading.Tasks.Extensions (>= 4.5.2) (via package/Azure.Identity 1.0.0), System.Threading.Tasks.Extensions (>= 4.5.1) (via package/Moq 4.10.1), System.Threading.Tasks.Extensions (>= 4.5.2) (via project/Azure.Core 1.0.0), System.Threading.Tasks.Extensions (>= 4.5.2) (via project/Azure.Security.KeyVault.Secrets 4.0.0), System.Threading.Tasks.Extensions (>= 4.5.1) (via package/Microsoft.AspNetCore.Server.Kestrel.Core 2.2.0), System.Threading.Tasks.Extensions (>= 4.3.0) (via package/System.Xml.ReaderWriter 4.3.0), System.Threading.Tasks.Extensions (>= 4.3.0) (via package/System.Xml.ReaderWriter 4.3.0) Framework: (.NETCoreApp,Version=v2.1)
                     D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj : error NU1106: Unable to satisfy conflicting requests for 'System.Buffers': System.Buffers (>= 4.5.0) (via project/Azure.Core 1.0.0), System.Buffers (>= 4.5.0) (via package/Azure.Core 1.0.0), System.Buffers (>= 4.5.0) (via package/Microsoft.Net.Http.Headers 2.2.0), System.Buffers (>= 4.5.0) (via package/Microsoft.AspNetCore.Http.Extensions 2.2.0), System.Buffers (>= 4.5.0) (via package/Microsoft.Net.Http.Headers 2.2.0) Framework: (.NETCoreApp,Version=v2.1)
                     D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj : error NU1106: Unable to satisfy conflicting requests for 'System.Diagnostics.DiagnosticSource': System.Diagnostics.DiagnosticSource (>= 4.6.0) (via project/Azure.Core 1.0.0), System.Diagnostics.DiagnosticSource (>= 4.5.0) (via package/Microsoft.AspNetCore.Hosting 2.2.0), System.Diagnostics.DiagnosticSource (>= 4.6.0) (via package/Azure.Core 1.0.0) Framework: (.NETCoreApp,Version=v2.1)
                     D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj : error NU1106: Unable to satisfy conflicting requests for 'System.Numerics.Vectors': System.Numerics.Vectors (>= 4.5.0) (via project/Azure.Core 1.0.0), System.Numerics.Vectors (>= 4.5.0) (via package/Microsoft.AspNetCore.Server.Kestrel.Core 2.2.0), System.Numerics.Vectors (>= 4.5.0) (via package/Azure.Core 1.0.0) Framework: (.NETCoreApp,Version=v2.1)
                     c:\program files\dotnet\sdk\3.0.100\NuGet.targets(123,5): error : An item with the same key has already been added. Key: (Azure.Core, 1.0.0) [D:\adp\azure-sdk-for-net\sdk\core\Azure.Core\tests\Azure.Core.Tests.csproj]
```

Reply:
We may need to file `An item with the same key has already been added.` on nuget, it's a `dict.Add` exception not something customers should see.

Reply:
@pakrm looks like we are hitting https://github.com/NuGet/Home/issues/6795. This fixes it though and so I'll keep this fix.